src/gluonts/shell/sagemaker/__init__.py

```diff
@@ -36,18 +37,16 @@ def __init__(self, path="/opt/ml"):
         self.datasets = self._get_datasets()
         self._check_sf2()
 
-    def _get_datasets(self):
-        freq = self.hyperparameters["time_freq"]
+    def _get_datasets(self) -> Dict[str, FileDataset]:
+        freq = self.hyperparameters["freq"]
```

Comment:
So this won't break anything?

Reply:
Good catch. I cleaned up all SageMaker-related configuration and environment bootstrapping logic (now all in `sagemaker/__init__.py`) and updated the PR.

The following method now ensures backwards-compatibility w.r.t. the `hyperparameters.json` file. Backwards-compatibility in the `metadata.json` file was already added in the current version.

```python
def _load_hyperparameters(path: MLPath, channels) -> dict:
    with path.hyperparameters.open() as json_file:
        hyperparameters = parse_sagemaker_parameters(json.load(json_file))

        for old_freq_name in ['time_freq', 'time_granularity']:
            if old_freq_name in hyperparameters:
                hyperparameters['freq'] = hyperparameters[old_freq_name]

        if "metadata" in channels:
            with (channels["metadata"] / "metadata.json").open() as file:
                metadata = MetaData(**json.load(file))
                hyperparameters.update(freq=metadata.freq)

        return hyperparameters
```